### PR TITLE
Add uniform random operator

### PR DESCRIPTION
**Pull request by reyoung**
It can be run both CPU/GPU. configure attributes are:

* min: the min value of uniform random
* max: the max value of uniform random
* dims: the dimension of output tensor
* seed: the random seed of uniform random. 0 means generate a seed each
        time.

### CODE REVIEW

File: paddle/framework/CMakeLists.txt [hunk 1]
```diff
@@ -38,9 +38,10 @@ cc_test(backward_test SRCS backward_test.cc DEPS backward)
 cc_library(paddle_pybind SHARED
     SRCS pybind.cc
     DEPS pybind python backward
-	fc_op
```
Comment (reyoung) [inline paddle/framework/CMakeLists.txt]: Use `space` for indent.

File: paddle/operators/uniform_random_op.cc [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      : OpProtoAndCheckerMaker(proto, op_checker) {
+    AddOutput("Out", "The output tensor of uniform random op");
+    AddComment(R"DOC(Uniform random operator.
+
+Used to initialize tensor with uniform random generator.
+)DOC");
+    AddAttr<std::vector<int>>("dims", "the dimension of random tensor");
+    AddAttr<float>("min", "Minimum value of uniform random").SetDefault(-1.0f);
+    AddAttr<float>("max", "Maximun value of uniform random").SetDefault(1.0f);
+    AddAttr<int>("seed",
```
Comment (dzhwinter) [inline paddle/operators/uniform_random_op.cc]: can we bind the `seed` to the device context or environment? 
In my view, the user may set a random seed before doing any experiment. He will not set a random seed in every random operator function calls.

File: paddle/operators/uniform_random_op.cc [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                 "Random seed of uniform random. "
```
Comment (dzhwinter) [inline paddle/operators/uniform_random_op.cc]: further more, each stream in GPU needs a different seed.

File: paddle/operators/uniform_random_op.cc [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License. */
+
+#include "paddle/operators/uniform_random_op.h"
+
+namespace paddle {
+namespace operators {
+class RandomOp : public OperatorWithKernel {
```
Comment (dzhwinter) [inline paddle/operators/uniform_random_op.cc]: `RamdomOp` is too general, maybe rename with `UniformRandomOp` is better?  I have implemented a Gaussian random op.

File: paddle/operators/uniform_random_op.cc [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ protected:
+  void InferShape(const InferShapeContext &ctx) const override {
+    PADDLE_ENFORCE(GetAttr<float>("min") < GetAttr<float>("max"),
+                   "uniform_random's min must less then max");
+    auto tensor = ctx.Output<Tensor>(0);
```
Comment (dzhwinter) [inline paddle/operators/uniform_random_op.cc]: rename `auto` with`auto*` for clear.

File: paddle/operators/uniform_random_op.cc (hunk 4 above)
Comment (reyoung) [inline paddle/operators/uniform_random_op.cc]: OK.

File: paddle/operators/uniform_random_op.cc (hunk 2 above)
Comment (reyoung) [inline paddle/operators/uniform_random_op.cc]: I think that logic should be handled in a higher level of Paddle. 

In normal situation, user does not need set seed, just set them to zero, Paddle will generate a seed from `std::random_device`

File: paddle/operators/uniform_random_op.cc (hunk 3 above)
Comment (reyoung) [inline paddle/operators/uniform_random_op.cc]: Each stream in GPU's seed is decided outside operator.

File: paddle/operators/uniform_random_op.cc (hunk 5 above)
Comment (reyoung) [inline paddle/operators/uniform_random_op.cc]: Done.

File: paddle/operators/uniform_random_op.cc (hunk 4 above)
Comment (reyoung) [inline paddle/operators/uniform_random_op.cc]: Done.

File: paddle/operators/uniform_random_op.cu [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+#include <thrust/random.h>
+#include <thrust/transform.h>
+#include "paddle/operators/type_alias.h"
+
+namespace paddle {
+namespace operators {
+
+template <typename T>
+struct UniformGenerator {
+  T min_, max_;
```
Comment (reyoung) [inline paddle/operators/uniform_random_op.cu]: Done. TODO is added.

In Eigen, only `normal` and `uniform` two generators are supported. i.e., Eigen is not good for randomizing a tensor.

File: paddle/operators/uniform_random_op.cc [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    }
+  }
+};
+
+class UniformRandomOp : public framework::OperatorWithKernel {
+ protected:
+  void InferShape(const framework::InferShapeContext& ctx) const override {
+    PADDLE_ENFORCE(GetAttr<float>("min") < GetAttr<float>("max"),
+                   "uniform_random's min must less then max");
+    auto tensor = ctx.Output<framework::Tensor>(0);
```
Comment (dzhwinter) [inline paddle/operators/uniform_random_op.cc]: use `auto *` instead of `auto` for clear.

Comment (reyoung) [inline paddle/operators/uniform_random_op.cc]: Done.

**Pull request merged**